addons/purchase/models/purchase.py

```diff
@@ -924,10 +924,7 @@ def write(self, values):
                                                          subtype_id=self.env.ref('mail.mt_note').id)
         if 'qty_received' in values:
             for line in self:
-                if values['qty_received'] != line.qty_received and line.order_id.state == 'purchase':
-                    line.order_id.message_post_with_view('purchase.track_po_line_qty_received_template',
-                                                         values={'line': line, 'qty_received': values['qty_received']},
-                                                         subtype_id=self.env.ref('mail.mt_note').id)
+                line._track_qty_received(values['qty_received'])
```

Comment:
Nice fix,
but you call `_track_qty_received` in purchase, but the definition of the method is in purchase_stock.

Reply:
Je ne vois pas de quoi tu parle

Reply:
![](https://media1.giphy.com/media/vXwhrRaKPPaYtkSOyO/giphy.gif)